### DISCUSSION TITLE
[Question][Hw5] cmp of Sign

### DISCUSSION THREAD

**Discussion by esk223** [prosyslab-classroom/cs348-information-security#587] [Q&A]
I'm a bit confused about cmp function of Sign.
1) I understood (cmp < Neg Pos) is always 1 which represents true. Thus, should it return Pos?
2) Similarly (cmp < Pos Neg) is always false, so should it return Zero?
3) (cmp < Pos Pos) can be either true or false. In this case, is the result Top?
4) Do (cmp _ Bot _) and (cmp _ _ Bot) always return Bot?

**Comment by KihongHeo** [comment]
Designing such operators is also a part of this homework.
Think about the principle more carefully:
1. the output of abstract operator should always subsume that of concrete operator.
2. the output should be also as precise as possible.
